images/rag/embedding.py

```diff
@@ -38,73 +41,70 @@ def get_eval_results(key, eval_results):
-        
-    elif args.vector_type == "chromadb":
-        print("** chromadb embeddings ")
-        #Validate Inputs 
-        if args.url is None or args.port is None: 
-            print(" Missing URL or PORT ")
-            return 
-        
-        chroma_client = chroma.HttpClient(host=args.url, port=args.port, headers=json.loads(args.auth) )
-        collection = chroma_client.create_collection(
-            name=args.collection_name,
-            get_or_create = True            
-            )
-        vector_store = ChromaVectorStore(chroma_collection=collection)
-        storage_context = StorageContext.from_defaults(vector_store=vector_store)
-        
-        
-    elif args.vector_type == "chromadb-local":
-        chroma_client = chromadb.Client()
-        collection = chroma_client.create_collection(
-            name=args.collection_name,
-            get_or_create = True            
+
+    # setup storage context
+    match args.vector_type:
+        case "local":
+            print("** Local embeddings")
+            storage_context = StorageContext.from_defaults()
+        case "chromadb":
+            print("** chromadb embeddings")
+            #Validate Inputs
+            if args.url is None or args.port is None:
+                print("Missing URL or PORT")
+                return
+            chroma_client = chromadb.HttpClient(host=args.url, port=args.port, headers=json.loads(args.auth))
+            collection = chroma_client.create_collection(
+                name=args.collection_name,
+                get_or_create = True
+                )
+            vector_store = ChromaVectorStore(chroma_collection=collection)
+            storage_context = StorageContext.from_defaults(vector_store=vector_store)
+        case "chromadb-local":
+            chroma_client = chromadb.Client()
+            collection = chroma_client.create_collection(
+                name=args.collection_name,
+                get_or_create = True
             )
-        vector_store = ChromaVectorStore(chroma_collection=collection)
-        storage_context = StorageContext.from_defaults(vector_store=vector_store )
-    
-        
-    elif args.vector_type == "milvus":
-        return 
-    
+            vector_store = ChromaVectorStore(chroma_collection=collection)
+            storage_context = StorageContext.from_defaults(vector_store=vector_store )
+        case "faiss":
+            faiss_index = faiss.IndexFlatL2(768)
+            vector_store = FaissVectorStore(faiss_index=faiss_index)
+            storage_context = StorageContext.from_defaults(vector_store=vector_store)
+
     print("** Configured storage context")
-    
-    service_context = ServiceContext.from_defaults(chunk_size=CHUNK_SIZE, chunk_overlap=CHUNK_OVERLAP,embed_model=args.model, llm='local')
+
+    service_context = ServiceContext.from_defaults(chunk_size=CHUNK_SIZE, chunk_overlap=CHUNK_OVERLAP,embed_model=args.model, llm=None)
```

Comment:
Changing llm to None triggered a MockLLM on my computer.
i am not sure what are the implication

Reply:
LlamaIndex assumes by default it's going to be talking to a real/big LLM, and that's what that parameter is referring to. Since we're only generating embeddings, we don't need the big LLM and can avoid loading it entirely.